api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml

```diff
@@ -1658,6 +1659,32 @@ stages:
           total_affected_items: 0
           total_failed_items: 1
 
+  # Agent assignment below requires the agent not to be in the group to succeed
```

Comment:
Perhaps not the most efficient solution but it was the only one that worked, tried modifying previous tests to change the state of the cluster when it reached the failing test but every tiny change caused other tests to fail. This was the simplest way to solve the issue.